src/service/fixed-layer.js

```diff
@@ -686,116 +856,69 @@ export class FixedLayer {
-    } else {
-      fe.placeholder.parentElement.removeChild(fe.placeholder);
-    }
-  }
 
+/**
+ * The fixed layer is created inside the shadow root of the `<body>` element
+ * and fixed elements are distributed into this element via slots.
+ * @implements {TransferLayerDef}
+ */
+class TransferLayerShadow {
   /**
-   * @return {?Element}
+   * @param {!Document} doc
    */
-  getTransferLayer_() {
-    // This mode is only allowed for a single-doc case.
-    if (!this.transfer_ || this.transferLayer_) {
-      return this.transferLayer_;
-    }
-    const doc = this.ampdoc.win.document;
-    this.transferLayer_ = doc.body.cloneNode(/* deep */ false);
-    this.transferLayer_.removeAttribute('style');
-    setStyles(this.transferLayer_, {
+  constructor(doc) {
+    /** @private @const {!Document} */
+    this.doc_ = doc;
+
+    /** @private @const {!Element} */
+    this.layer_ = doc.createElement('div');
+    this.layer_.id = 'i-amphtml-fixed-layer';
+    setImportantStyles(this.layer_, {
       position: 'absolute',
       top: 0,
       left: 0,
       height: 0,
       width: 0,
-      pointerEvents: 'none',
       overflow: 'hidden',
-
-      // Reset possible BODY styles.
-      animation: 'none',
-      background: 'none',
-      border: 'none',
-      borderImage: 'none',
-      boxSizing: 'border-box',
-      boxShadow: 'none',
-      display: 'block',
-      float: 'none',
-      margin: 0,
-      opacity: 1,
-      outline: 'none',
-      padding: 'none',
-      transform: 'none',
-      transition: 'none',
-      visibility: 'visible',
     });
-    doc.documentElement.appendChild(this.transferLayer_);
-    return this.transferLayer_;
+    doc.body.shadowRoot.appendChild(this.layer_);
   }
 
-  /**
-   * Find all `position:fixed` and `position:sticky` elements.
-   * @param {!Array<CSSRule>} rules
-   * @param {!Array<string>} foundSelectors
-   * @param {!Array<string>} stickySelectors
-   * @private
-   */
-  discoverSelectors_(rules, foundSelectors, stickySelectors) {
-    for (let i = 0; i < rules.length; i++) {
-      const rule = rules[i];
-      if (rule.type == /* CSSStyleRule */ 1) {
-        if (rule.selectorText != '*' && rule.style.position) {
-          if (rule.style.position == 'fixed') {
-            foundSelectors.push(rule.selectorText);
-          } else if (endsWith(rule.style.position, 'sticky')) {
-            stickySelectors.push(rule.selectorText);
-          }
-        }
-      } else if (rule.type == /* CSSMediaRule */ 4) {
-        this.discoverSelectors_(rule.cssRules, foundSelectors, stickySelectors);
-      } else if (rule.type == /* CSSSupportsRule */ 12) {
-        this.discoverSelectors_(rule.cssRules, foundSelectors, stickySelectors);
-      }
-    }
+  /** @override */
+  setVisible(visible) {
+    setStyle(this.layer_, 'visibility', visible ? 'visible' : 'hidden');
   }
-}
 
+  /** @override */
+  update() {
+    // Nothing to do.
```

Comment:
Do we not need to do the same non-inherited CSS syncing?

Reply:
No, it's now the responsibility of the viewport, not the transfer layer.